src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,6 +219,41 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
             }
         }
 
+        private async Task LoadInitialConfiguration()
+        {
+            // Check if this is the first attempt to load the configuration for failed startup load as a part of refresh
+            if (MinCacheExpirationTime == null)
+            {
+                IEnumerable<KeyValueWatcher> watchers = _options.ChangeWatchers.Union(_options.MultiKeyWatchers);
+
+                if (!watchers.Any())
+                {
+                    // No registration was found for refresh
+                    return;

```

Comment:
>return [](start = 20, length = 6)

If the code returns here then we never mark any state to prevent it from happening every time refresh is called. Intended?